### DISCUSSION TITLE
Getting error when build edk2 environemnt

### DISCUSSION THREAD

**Discussion by tushar95mahawar** [tianocore/edk2#3668] [General]
I am using EDK2 and visual studio 2015. i followed the set up to build edk2 environment on windows system. After sending following command " build " on windows command line. I got the  following message.

```
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 14.0\Vc\bin\x86_amd64\cl.exe"' : return code '0x2'
Stop.

build.py...
 : error 7000: Failed to execute command
        C:\Program Files (x86)\Microsoft Visual Studio 14.0\Vc\bin\nmake.exe /nologo tbuild [c:\edkii\Build\MdeModule\DEBUG_VS2015x86\X64\MdeModulePkg\Library\BrotliCustomDecompressLib\BrotliCustomDecompressLib]

build.py...
 : error 7000: Failed to execute command
        C:\Program Files (x86)\Microsoft Visual Studio 14.0\Vc\bin\nmake.exe /nologo tbuild [c:\edkii\Build\MdeModule\DEBUG_VS2015x86\IA32\MdeModulePkg\Library\BrotliCustomDecompressLib\BrotliCustomDecompressLib]

build.py...
 : error F002: Failed to build module
        c:\edkii\MdeModulePkg\Library\BrotliCustomDecompressLib\BrotliCustomDecompressLib.inf [X64, VS2015x86, DEBUG]

- Failed -
Build end time: 14:01:05, Nov.28 2022
Build total time: 00:00:15
```

**Comment by lersi** [comment]
can you please provide the error message that `cl.exe` gave?
because I only see here the error message from NMAKE which doesn't support diagnosing the problem.

by the way, I made a project that helps you build more easily, and such problems should not happen there.
project link: [edk2_project_template](https://github.com/lersi/edk2_project_template)

**Reply by tushar95mahawar** [reply]
The following error message is showing.
```
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 14.0\Vc\bin\x86_amd64\cl.exe"' : return code '0x2'
Stop.
Building ... c:\edkii\MdePkg\Library\UefiUsbLib\UefiUsbLib.inf [X64]
```

**Reply by tushar95mahawar** [reply]
by the way, build is successful is showing  when i ACTIVE_PLATFORM   = EmulatorPkg/EmulatorPkg.dsc
But when I set ACTIVE_PLATFORM   = MdeModulePkg/MdeModulePkg.dsc and pass build command it showing error any solution for this
